File: scratch/download_gdrive.py
import urllib.request
import re

file_id = '1WXA468KXhivdZq01BMOuwtp2AqgD3zXF'
url = f'https://drive.google.com/uc?export=download&id={file_id}'

req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
try:
    with urllib.request.urlopen(req) as response:
        content = response.read()
        
        # Check if Google Drive sent a confirmation page
        content_str = content.decode('utf-8', errors='ignore')
        confirm_match = re.search(r'confirm=([0-9a-zA-Z_]+)', content_str)
        if confirm_match:
            confirm_code = confirm_match.group(1)
            confirm_url = f'https://drive.google.com/uc?export=download&confirm={confirm_code}&id={file_id}'
            print(f'Downloading with confirmation code: {confirm_code}')
            req2 = urllib.request.Request(confirm_url, headers={'User-Agent': 'Mozilla/5.0'})
            with urllib.request.urlopen(req2) as resp2:
                content = resp2.read()

        with open('d:/Institute_Project/public/glb/logo.glb', 'wb') as f:
            f.write(content)
        print(f'Downloaded {len(content)} bytes successfully to public/glb/logo.glb')
except Exception as e:
    print(f'Error: {e}')

File: scratch/download_gdrive.js
const fs = require('fs');
const https = require('https');

const fileId = '1WXA468KXhivdZq01BMOuwtp2AqgD3zXF';
const url = `https://drive.usercontent.google.com/download?id=${fileId}&export=download&confirm=t`;

console.log(`Downloading GLB from Google Drive (${fileId})...`);

function fetchFile(targetUrl) {
  https.get(targetUrl, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
    if (res.statusCode === 301 || res.statusCode === 302) {
      console.log(`Redirecting to: ${res.headers.location}`);
      return fetchFile(res.headers.location);
    }
    if (res.statusCode === 200) {
      const file = fs.createWriteStream('d:/Institute_Project/public/glb/logo.glb');
      res.pipe(file);
      file.on('finish', () => {
        file.close();
        const size = fs.statSync('d:/Institute_Project/public/glb/logo.glb').size;
        console.log(`Successfully downloaded GLB model! File size: ${size} bytes`);
      });
    } else {
      console.error(`Failed with HTTP status code: ${res.statusCode}`);
    }
  }).on('error', (err) => {
    console.error('Error downloading:', err.message);
  });
}

fetchFile(url);
